src/app_service/service/community/service.nim

```diff
@@ -1478,7 +1479,7 @@ QtObject:
         slot: "onAsyncRequestToJoinCommunityDone",
         communityId: communityId,
         ensName: ensName,
-        password: password,
+        password: if password != "": utils.hashPassword(password) else: "",
```

Comment:
Do you think we could hash it even earlier? Like in the module where we get it?

Reply:
Of course we could. I think doing it here is a good trade-off between code duplication and proneness to arguments being logged. We don't do any logging at service level at the moment.

By the way, do you believe that a hashed password is significantly safer than plain text in this exact scenario? Even if a potential attacker reads a hashed password from our logs, they can still use it to log in. That being said, I believe the ultimate solution lies in log levels. Information like function arguments, which might contain passwords (whether in plain-text or hashed), should be at the debug level. Users should be warned about the potential risks before enabling debug-level logging.

Reply:
Issue for the popup: https://github.com/status-im/status-desktop/issues/12308